sdk/src/main/scala/com/horizen/account/abi/ABIDecoder.scala

```diff
@@ -0,0 +1,42 @@
+package com.horizen.account.abi
+
+import org.web3j.abi.datatypes.Type
+import org.web3j.abi.{DefaultFunctionReturnDecoder, TypeReference}
+
+import java.util
+import scala.util.Try
+
+object ABIDecoder {
+  val OP_CODE_LENGTH = 4
+
+  def getArgumentsFromData(data: Array[Byte]): Array[Byte] = {
+    if (data.length < ABIDecoder.OP_CODE_LENGTH) throw new IllegalArgumentException("Data length " + data.length + " must be >= " + ABIDecoder.OP_CODE_LENGTH)
+    util.Arrays.copyOfRange(data, ABIDecoder.OP_CODE_LENGTH, data.length)
+  }
+
+  def getOpCodeFromData(data: Array[Byte]): Array[Byte] = {
+    if (data.length < ABIDecoder.OP_CODE_LENGTH) throw new IllegalArgumentException("Data length " + data.length + " must be >= " + ABIDecoder.OP_CODE_LENGTH)
+    util.Arrays.copyOf(data, ABIDecoder.OP_CODE_LENGTH)
+  }
+
+}
+
+trait ABIDecoder[T] {
+  private val decoder = new DefaultFunctionReturnDecoder()
```

Comment:
let's remove the val here and create in the decode method on demand.
I dan't see it useful to keep here

Reply:
Done